tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/anno/AnnotationTestBean.java

```diff
@@ -0,0 +1,142 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import ee.jakarta.tck.concurrent.common.context.IntContext;
+import ee.jakarta.tck.concurrent.common.context.StringContext;
+import jakarta.ejb.EJBException;
+import jakarta.ejb.Local;
+import jakarta.ejb.Stateless;
+import jakarta.enterprise.concurrent.ManagedScheduledExecutorService;
+import jakarta.enterprise.concurrent.ManagedThreadFactory;
+
+@Local(AnnotationTestBeanInterface.class)
+@Stateless
+public class AnnotationTestBean implements AnnotationTestBeanInterface {
+    private static final long MAX_WAIT_SECONDS = TimeUnit.MINUTES.toSeconds(2);
+
+    @Override
+    public void testAnnotationDefinesManagedScheduledExecutor() {
+        try {
+            LinkedBlockingQueue<Integer> started = new LinkedBlockingQueue<Integer>();
+            CountDownLatch taskCanEnd = new CountDownLatch(1);
+
+            Callable<String> task = () -> {
+                started.add(IntContext.get());
+                assertTrue(taskCanEnd.await(MAX_WAIT_SECONDS, TimeUnit.SECONDS));
+                return StringContext.get();
+            };
+
+            ManagedScheduledExecutorService executor = InitialContext
+                    .doLookup("java:global/concurrent/ScheduledExecutorE");
+
+            IntContext.set(3000);
+
+            StringContext.set("testAnnotationDefinesManagedScheduledExecutor-1");
+            Future<String> future1 = executor.submit(task);
+
+            StringContext.set("testAnnotationDefinesManagedScheduledExecutor-2");
+            Future<String> future2 = executor.submit(task);
+
+            StringContext.set("testAnnotationDefinesManagedScheduledExecutor-3");
+            Future<String> future3 = executor.submit(task);
+
+            StringContext.set("testAnnotationDefinesManagedScheduledExecutor-4");
+
+            // 2 can start per max-async
+            assertEquals(started.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(0));
+            assertEquals(started.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(0));
+            assertEquals(started.poll(1, TimeUnit.SECONDS), null);
+
+            taskCanEnd.countDown();
+
+            assertEquals(future1.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS),
+                    "testAnnotationDefinesManagedScheduledExecutor-1");
+            assertEquals(future2.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS),
+                    "testAnnotationDefinesManagedScheduledExecutor-2");
+            assertEquals(future3.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS),
+                    "testAnnotationDefinesManagedScheduledExecutor-3");
```

Comment:
more reversed assertions?

```suggestion
            assertEquals("testAnnotationDefinesManagedScheduledExecutor-1",
                    future1.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS));
            assertEquals("testAnnotationDefinesManagedScheduledExecutor-2",
                    future2.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS));
            assertEquals("testAnnotationDefinesManagedScheduledExecutor-3",
                    future3.get(MAX_WAIT_SECONDS, TimeUnit.SECONDS));
```